### PR TITLE
Snapshotter status report structure update and support for http IPFS URL

### PR DESCRIPTION
**Pull request by swagftw**
Includes changes from #34, #35, #37 and #38

### Checklist
- [x] My branch is up-to-date with upstream/main branch.
- [x] Everything works and tested for major version of Python/NodeJS/Go and above.
- [x] I ran pre-commit checks against my changes.

### Summary
- Snapshotter status report now has valid `reason` for failure and `snapshot data` for better tracing
- Support for http IPFS URL has been added, eg. `http://localhost:5001` or `https://ipfs.infura.io:5001` are now valid URLs
- Signature deadline has been increased from `10` to `30` in `settings.example.com`
- Payload commit service now publishes `snapshot submmited` event. For more info refer #37
- RabbitMQ channel leak issue fixed in #38
- Update Protocol State default contract to `0x19B7f6c458bBd6faDc36D39Ae77B132CC3b5B159`

## Deployment Instructions
Increase Signature deadline (`settings.signer.deadlineBuffer` key) to `30`
Refer PR #37 for config changes related to RabbitMQ

### CODE REVIEW

**Review by anomit — APPROVED** [review]
LGTM

**Review by xadahiya — CHANGES_REQUESTED** [review]
Noticed these errors upon running it for some time. Please take a look
Screenshot 2023-06-13 at 11 30 17 PM

**Comment by swagftw** [comment]
> Noticed these errors upon running it for some time. Please take a look

Issue has been addressed in the PR #38

File: snapshotter_autofill.sh [hunk 1]
```diff
@@ -54,8 +54,8 @@ export ipfs_url="${IPFS_URL:-/dns/ipfs/tcp/5001}"
 export ipfs_api_key="${IPFS_API_KEY:-}"
 export ipfs_api_secret="${IPFS_API_SECRET:-}"
 
-export protocol_state_contract="${PROTOCOL_STATE_CONTRACT:-0xb1983E8ce013F62fC503AA5C7704B91a3bf25eC8}"
-export relayer_host="${RELAYER_HOST:-https://relayer-prod1b.powerloom.io/75822d76fa4d497ab3b409b3f097f4fa}"
+export protocol_state_contract="${PROTOCOL_STATE_CONTRACT:-0x19B7f6c458bBd6faDc36D39Ae77B132CC3b5B159}"
+export relayer_host="${RELAYER_HOST:-https://relayer-test1b.powerloom.io}"
```
Comment (xadahiya) [inline snapshotter_autofill.sh:58]: @swagftw?? check default relayer

**Review by xadahiya — APPROVED** [review]
New changes seem to have fixed the Rabbitmq channel issues. New code is running without any issues in stg instances for some time now. Approving

Screenshot 2023-06-14 at 1 36 35 PM

**Pull request merged**